src/Core/Models/Date.php

```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.
+ * Licensed under the MIT License.  See License in the project root
+ * for license information.
+ */
+
+namespace Microsoft\Graph\Core\Models;
+
+class Date extends \DateTime implements \JsonSerializable {
```

Comment:
Don't think we should extend `\DateTime`. It helps us get away with a non-breaking change but it's a false inheritance relationship since methods on a [DateTime object](https://www.php.net/manual/en/class.datetime) don't apply here... 

I suggest we accept a year, month & date in the constructor & support a static method to create a Date from a `\DateTime` etc

[.NET SDK](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/dev/src/Microsoft.Graph.Core/Models/Date.cs) has a good reference implementation

Reply:
There is a method in DateTime that already allows us to set date from the three parameters ```php setDate($day, $month, $year)```

Reply:
It makes sense in this case to be able to create a date object from DateTime without specifying the format.

Reply:
My concern is that inheritance insinuates that this Date type has ALL the behaviour & properties of a DateTime. From the  Edm.Date definition, which this object should represent, there's no use case for some inherited behaviour like `setTime(), setTimestamp(), setTimezone()`.

I think this prevents Date from being a candidate to inherit from DateTime

Reply:
There are things done in the DateTime class that will take too much code to handle on the new implementation which would be unnecessary in this case. For instance how we handle leap years for Date class and how to handle things like timezone in the TimeOfDay class. Do you have any undefined behaviour you might have noticed with the implementation?

Reply:
I think we should focus on creating a model that matches the properties and functionality defined for Edm.Date & Edm.TimeOfDay in the Odata spec.

So no need for handling timezones, leap yrs, adding/subtracting intervals etc.

Reply:
Undefined behaviour for the current Date implementation is the extra time-related methods provided in a DateTime that don't apply to an Edm.Date
And vice versa for a TimeOfDay extending DateTime

Reply:
> I think we should focus on creating a model that matches the properties and functionality defined for Edm.Date & Edm.TimeOfDay in the Odata spec.
> 
> So no need for handling timezones, leap yrs, adding/subtracting intervals etc.

Just a container class to hold a year, month & day for a Date that can do some serialization/deserialization
For a TimeOfDay: hr, min, sec, serialize/deserialize